pages/product/[productId]/index.tsx

```diff
@@ -0,0 +1,47 @@
+import { Panel, Text } from "@bigcommerce/big-design";
+import { useRouter } from "next/router";
+import ErrorMessage from "@components/error";
+import Loading from "@components/loading";
+import { useProductInfo, useProductList } from "@lib/hooks";
+
+const SidePanel = () => {
+  const router = useRouter();
+  const productId = Number(router.query?.productId);
+  const { error, isLoading, list = [] } = useProductList();
+  const { isLoading: isInfoLoading, product } = useProductInfo(productId, list);
+  const {
+    description,
+    is_visible: isVisible,
+    name,
+    price,
+    type,
+  } = product ?? {};
+  const panelData = { description, isVisible, name, price, type };
+
+  const capitalizeFirst = (str: string) => {
+    return str.charAt(0).toUpperCase() + str.slice(1);
+  };
```

Comment:
generally, I like having helper functions, though if we're only using a single line function in one place, it makes me wonder if it's necessary... we might just assign this to a variable

```suggestion
    const productType = type.replace(/^\w/, c => c.toUpperCase());
```